src/coreclr/jit/assertionprop.cpp

```diff
@@ -1149,16 +1149,14 @@ AssertionIndex Compiler::optCreateAssertion(GenTree*         op1,
             }
         }
 
-        if (fgIsBigOffset(offset) || op1->gtOper != GT_LCL_VAR)
+        if (fgIsBigOffset(offset) || !op1->IsLocal())
```

Comment:
Is this change ok? `LCL_FLD` will create an assertion about the parent local without encoding any information about the offset, which seems odd.